### ISSUE TITLE
 Generates Python functions from LaTeX math description?

### ISSUE THREAD

**Issue by ghosthamlet** [google/latexify_py#26]
Is it harder to generate Python functions from LaTeX math description? This function will be great for read ML papers.

**Comment by odashi** [comment]
Yes I think so, because LaTeX math is just a markup for printing which is more general than programming languages. It would be great if we have such options, but I guess we need some additional effort, such as:

- Add an ML (!) pipeline to convert LaTeX source to Python. This would be a kind of recent research orientation of transpilers (I have read some papers on ML conferences).
- Defines some restricted language on LaTeX math so that it can be converted into Python directly.

**Comment by liudonghua123** [comment]
If this feature is available, a lot of students, teachers, researchers would embrace python, especially `latexify`.

Looking forward to this nice feature.

Maybe `compile`/`exec` used to implement it.

```python
new_func='def next_element(x):\n  return x+1'
the_code=compile(new_func,'test','exec')
exec(the_code)
next_element(1)
```
> https://stackoverflow.com/questions/11291242/python-dynamically-create-function-at-runtime

**Comment by odashi** [comment]
This is a general discussion and not applicable for the current purpose of this library (python -> latex). Close it for now.

**Comment by ghosthamlet** [comment]
Maybe closing now, 
galactica model Can translate Math formula to plain english, 
https://galactica.org/explore/ (See `Scientific from Scratch` section),
and generate code of a function from plain english have many options: copilot, codex.

**Comment by odashi** [comment]
Great. I have also tried some LM models learnt form natural language and code several months ago (related to my research work), and found that they possibly could translate between them. And now we could find some services based on similar techniques. Of course they are definitely ML models: the result is not entirely guaranteed, but I guessed it is very promising.

On the other hand, latexify is aiming to become a "compiler", which have to guarantee every result if the syntax is supported. But I'm also considering if some ML techniques can be applied to this library as well (e.g., compilers requrie optimization, and optimization often requires some search techniques, which can be reinforced by a strong suggestion system)
